### PR TITLE
Code Implementation to obtain users with login information

### PR DESCRIPTION
**Pull request by Gra2411**
Adding POST operation to login a user

### CODE REVIEW

File: src/main/java/com/agilethought/atsceapi/exceptions/BadRequestException.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.BAD_REQUEST)
+public class BadRequestException extends Exception{
+    public BadRequestException(String message) {
+        super(message);
+    }
+
+}
```
Comment (luisgalindoluna) [inline src/main/java/com/agilethought/atsceapi/exceptions/BadRequestException.java]: I don't think creating exception classes and methods is part of your user story. Feel free to backup this code for later use, but for now it should not be in here.

File: src/main/java/com/agilethought/atsceapi/exceptions/NotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(HttpStatus.NOT_FOUND)
+public class NotFoundException extends RuntimeException{
+    public NotFoundException(String message) {
+        super(message);
+    }
+}
```
Comment (luisgalindoluna) [inline src/main/java/com/agilethought/atsceapi/exceptions/NotFoundException.java]: Again, creating exception classes and methods is not a part of your user story.

File: src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java [hunk 3]
```diff
@@ -1,5 +1,74 @@
...
+import com.agilethought.atsceapi.model.User;
+import com.agilethought.atsceapi.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+
+import java.util.List;
+
```
Comment (luisgalindoluna) [inline src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java]: There's way too many empty spaces in all your classes, please go through EVERY SINGLE ONE of your classes and interfaces and hit "Ctrl+Shift+F." This will properly format your code.

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1 +0,0 @@
-
```
Comment (luisgalindoluna) [inline src/main/resources/application.properties:1]: What's going on here? I'm guessing you're trying to merge that hyphen. Delete it, this should not be showing up in your commit.

File: src/test/java/com/agilethought/atsceapi/ATSceApiApplicationTests.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package com.agilethought.atsceapi;
 
 import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
```
Comment (luisgalindoluna) [inline src/test/java/com/agilethought/atsceapi/ATSceApiApplicationTests.java]: Why did you add this? I believe this is also outside of the scope of your tasks, I think it's better if you delete it, this should also not appear in your commit

**Review by luisgalindoluna — CHANGES_REQUESTED** [review]
Please take a look at my comments

File: src/main/java/com/agilethought/atsceapi/exceptions/BadRequestException.java (hunk 1 above)
Comment (ivanflores15) [reply src/main/java/com/agilethought/atsceapi/exceptions/BadRequestException.java]: I think we can left this classes for future implementation, but this exception should be unauthorized for both cases

Comment (luisgalindoluna) [reply src/main/java/com/agilethought/atsceapi/exceptions/BadRequestException.java]: So you're saying to leave the classes themselves but remove the implementation for now?

Comment (ivanflores15) [reply src/main/java/com/agilethought/atsceapi/exceptions/BadRequestException.java]: As part of her Us is thrown 401 exception if data is invalid, she needs to change this classes for one that throws 401 instead of 404 or 400 codes.

Comment (luisgalindoluna) [reply src/main/java/com/agilethought/atsceapi/exceptions/BadRequestException.java]: Ah, I see got it.

File: pom.xml [hunk 6]
```diff
@@ -33,6 +33,45 @@
...
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-mongodb</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+		<dependency>
```
Comment (ivanflores15) [inline pom.xml]: `<dependency>
			<groupId>org.apiguardian</groupId>
			<artifactId>apiguardian-api</artifactId>
			<version>1.1.0</version>
		</dependency>

		<dependency>
			<groupId>io.swagger</groupId>
			<artifactId>swagger-core</artifactId>
			<version>1.5.4</version>
		</dependency>

		<dependency>
			<groupId>javax.ws.rs</groupId>
			<artifactId>javax.ws.rs-api</artifactId>
			<version>2.1.1</version>
		</dependency>`
This dependencies are never used, please remove

File: src/main/java/com/agilethought/atsceapi/controller/UserController.java [hunk 7]
```diff
@@ -1,5 +1,20 @@
...
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(value = "/at-sce-api/api/v1")
 public class UserController {
 
+  @Autowired
+  private UserService userService;
+
+  @PostMapping(value= "/login")
```
Comment (ivanflores15) [inline src/main/java/com/agilethought/atsceapi/controller/UserController.java]: There should be a space between each side of = sign, add an space.

File: src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java [hunk 8]
```diff
@@ -1,5 +1,51 @@
...
+    private static boolean isValid(String email) {
+        String regex = "^[\\w-_\\.+]*[\\w-_\\.]\\@([\\w]+\\.)+[\\w]+[\\w]$";
+        return email.matches(regex);
+    }
+
+    private static boolean isStringLowerCase(String str){
+        //convert String to char array
+        char[] charArray = str.toCharArray();
+        for(int i=0; i < charArray.length; i++){
```
Comment (ivanflores15) [inline src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java]: = sign, add spaces

**Review by ivanflores15 — CHANGES_REQUESTED** [review]

**Review by luisgalindoluna — APPROVED** [review]
Looks good to me. 
And for future reference keep your commit messages way shorter and to the point.

**Review by ivanflores15 — APPROVED** [review]

File: pom.xml [hunk 9]
```diff
@@ -33,6 +33,28 @@
...
 			</exclusions>
 		</dependency>
+
+		<dependency>
+			<groupId>org.projectlombok</groupId>
+			<artifactId>lombok</artifactId>
+			<version>1.18.18</version>
+		</dependency>
+
+		<dependency>
```
Comment (VictorMunguiaEnciso) [inline pom.xml]: Could you please delete this repeated lombok dependency ?

**Review by VictorMunguiaEnciso — CHANGES_REQUESTED** [review]

File: src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java [hunk 10]
```diff
@@ -1,5 +1,51 @@
...
+@Slf4j
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Override
+    public User loginMethod(LoginData loginData) {
+
+        if (isValid(loginData.getEmail()) && isStringLowerCase(loginData.getEmail())) {
```
Comment (VictorMunguiaEnciso) [inline src/main/java/com/agilethought/atsceapi/service/UserServiceImpl.java:22]: There are some validations in this class that we should move them to a specific class only for validations, but for now we are ok with them, it will be a further implementation in other sprint.

**Review by VictorMunguiaEnciso — APPROVED** [review]
